docs/hub/information_extraction.md

```diff
@@ -0,0 +1,103 @@
+# Customer Information Extraction
+
+In this guide, we'll walk through how to extract customer lead information using OpenAI's API and Pydantic. This use case is essential for seamlessly automating the process of extracting specific information from a context.
+
+If you want to try this out via `instructor hub`, you can pull it by running:
+
+```bash
+instructor hub pull --slug information_extraction --py > information_extraction.py
+```
+
+## Motivation
+
+You could potentially integrate this into a chatbot to extract relevant user information from user messages. With the use of machine learning driven validation it would reduce the need for a human to verify the information.
+
+## Defining the Structure
+
+We'll model a customer lead as a Lead object, including fields for the name and phone number. We'll use a Pydantic field_validator for data validation and formatting.
+
+## Extracting Lead Information
+
+To extract lead information, we create the `parse_lead_from_message` function which integrates Instructor. It calls OpenAI's API, processes the text, and returns the extracted lead information as a Lead object.
+
+## Evaluating Lead Extraction
+
+To showcase the `parse_lead_from_message` function we can provide sample user messages that may be obtained from a dialogue with a chatbot assistant.
+
+```python
+import instructor
+from openai import OpenAI
+from pydantic import BaseModel, field_validator
+
+
+class Lead(BaseModel):
+    name: str
+    phone_number: str
+
+    @field_validator("phone_number")
+    @classmethod
+    def validate_phone_number(cls, value):
```

Comment:
I think this can be rewritten as a better regex. check this out
https://docs.pydantic.dev/2.0/usage/types/extra_types/phone_numbers/

Reply:
Oh cool, yeah this seems like a better option